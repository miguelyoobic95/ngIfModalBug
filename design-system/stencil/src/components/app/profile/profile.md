---
name: Profile
category: App Components
---

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`config`|IProfileConfig|*required|contains the informations related to the user, the links and the logout|

```yoo-profile.html
<yoo-profile></yoo-profile>
```

```yoo-profile.js
let profile = document.querySelector('yoo-profile');
profile.config = {
  user: { firstName: 'Cecilia' , lastName: 'Adams' , role: 'Store Manager', imgData: 'https://res.cloudinary.com/www-yoobic-com/image/upload/w_1020,h_480,c_fill,q_auto:low,f_auto,fl_lossy/a_exif/v1522152485/k7dedr96ainrdhmkzlgj.png' }, // might have to change the names
  links: [{title: 'More',
    items: [
      {
        title: 'Documents',
        handler: () => {alert('Documents')}
      },
      {
        title: 'Contacts',
        handler: () => {alert('Contacts')}
      },
      {
        title: 'Notes',
        handler: () => {alert('Notes')}
      },
    ]},
    {title: 'Get started',
      items: [
        {
          title: 'What\'s New?',
          handler: () => {}
        },
        {
          title: 'Walkthrough',
          handler: () => {}
        }
      ]},
    {title: 'Support & Preferences',
      items: [
        {
          title: 'Settings',
          handler: () => {}
        },
        {
          title: 'Help Center',
          handler: () => {}
        },
        {
          title: 'Terms & Conditions',
          handler: () => {}
        }
      ]}
  ],
  logoutText: 'Log Out',
  hideLogout: false
};
```

```yoo-profile-modal.html
<yoo-button onclick="generateProfileModal()">
    <div class="icon">
        <i class="yo-user"></i>
    </div>
</yoo-button>
<yoo-modal-controller></yoo-modal-controller>
```

```yoo-profile-modal.css
    yoo-modal div.outer-container {
      width: 15rem !important;
      height: 20rem !important;
    }
````

```yoo-profile-modal.js
function generateProfileModal() {
    let modalCtrl = document.querySelector('yoo-modal-controller');
    let htmlElement = document.createElement('div');
    htmlElement.innerHTML = `<yoo-profile ></yoo-profile>`
    modalCtrl.generateModal({
      hasHeader: true,
      hasFooter: false,
      content: htmlElement,
      scrollEnabled: true,
    });
    modalCtrl.show();

    let profile = htmlElement.querySelector('yoo-profile');
    profile.config = {
      user: { firstName: 'Cecilia' , lastName: 'Adams' , role: 'Store Manager', imgData: 'https://res.cloudinary.com/www-yoobic-com/image/upload/w_1020,h_480,c_fill,q_auto:low,f_auto,fl_lossy/a_exif/v1522152485/k7dedr96ainrdhmkzlgj.png' }, // might have to change the names
      links: [{title: 'More',
        items: [
          {
            title: 'Documents',
            handler: () => {alert('Documents')}
          },
          {
            title: 'Contacts',
            handler: () => {alert('Contacts')}
          },
          {
            title: 'Notes',
            handler: () => {alert('Notes')}
          },
        ]},
        {title: 'Get started',
          items: [
            {
              title: 'What\'s New?',
              handler: () => {}
            },
            {
              title: 'Walkthrough',
              handler: () => {}
            }
          ]},
        {title: 'Support & Preferences',
          items: [
            {
              title: 'Settings',
              handler: () => {}
            },
            {
              title: 'Help Center',
              handler: () => {}
            },
            {
              title: 'Terms & Conditions',
              handler: () => {}
            }
          ]}
      ],
      logoutText: 'Log Out',
      hideLogout: false
    };
  }
```

## Events
|Attr|Description|
|---|---|---|
|`profileEdit`|event emitted when the edit button is clicked|
|`clicked`|event emitted when an item is clicked|
|`logout`|event emitted when logout is clicked|
