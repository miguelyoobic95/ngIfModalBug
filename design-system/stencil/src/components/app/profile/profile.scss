:host {
  height: 100%;
  font-size: 1.0625rem;
  font-weight: 400;

  .border {
    height: 1px;
    width: 100%;
    box-shadow: inset 0 -0.5px 0 0 rgba(173, 173, 173, 0.5);
  }

  .profile-content {
    height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    .profile-user {
      text-align: center;
      .profile-user-name {
        line-height: 1.3125rem;
      }
      .profile-user-role {
        line-height: 1rem;
        font-size: 0.8125rem;
        color: $stable;
      }
    }

    .profile-links {
      padding-left: 1rem;

      .profile-links-title {
        display: flex;
        align-items: center;
                color: $stable;
        width: 100%;
        height: 1.875rem;
        margin-top: 0.9375rem;
        font-size: 0.8125rem;
      }
      .profile-links-menu {

        list-style: none;
        padding: 0;
        margin:0;
        padding-bottom: 0.3125rem;
        .profile-links-menu-item {
          display: flex;
          justify-content: space-between;
          align-items: center;
          height: 2.75rem;
          font-weight: 400;
          cursor: pointer;
          .yo-right {
            padding-right: 1rem;
          }
        }
      }
    }

    .profile-logout {
      display: block;
      padding-left: 1rem;
      margin-top: 1.25rem;
      margin-bottom: 2.75rem;

      .profile-logout-text {
        display: flex;
        justify-content: space-between;
        align-items: center;
        color: $danger;
        height: 2.75rem;
      }
    }
  }

}
