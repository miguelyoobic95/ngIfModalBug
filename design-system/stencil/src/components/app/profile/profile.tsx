import { Component, Element, Event, EventEmitter, Prop, Listen } from '@stencil/core';

@Component({
  tag: 'yoo-profile',
  styleUrl: 'profile.scss',
  scoped: true
})
export class YooProfileComponent {

  @Prop() config: any;

  @Event() clicked: EventEmitter<string>;
  @Event() profileEdit: EventEmitter<boolean>;
  @Event() logout: EventEmitter<boolean>;

   @Element() host: HTMLStencilElement;

  @Listen('bottomRightClicked')
  onProfileEdit() {
    this.profileEdit.emit(true);
  }

  onClicked(item) {
    this.clicked.emit(item);
  }

  onLogout() {
    this.logout.emit(true);
  }

  render(): JSX.Element {
    return (
      <div class="profile-content">
        {(this.config ?
          <span>
            <div class="profile-user">
              <yoo-avatar onClick={() => this.onProfileEdit()} class="large gradient-success large-border" user={this.config.user} bottom-right-icon="yo-pen"></yoo-avatar>
              <div class="profile-user-name">{this.config.user.firstName} {this.config.user.lastName}</div>
              <div class="profile-user-role">{this.config.user.role}</div>
            </div>
            <div class="profile-links">
              {this.config.links.map(link => {
                return this.renderUl(link);
              })}
            </div>

            {(this.config.hideLogout ?
              null
              :
              <div class="profile-logout" onClick={() => this.onLogout()}>
                <div class="border"></div>
                <span class="profile-logout-text">{this.config.logoutText}</span>
                <div class="border"></div>
              </div>
            )}
          </span>
          :
          <div>loading</div>
        )}
      </div>
    );
  }

  renderUl(link: { title: string, items: { title: string, handler: () => void }[] }): JSX.Element {
    return (
      <span>
        <div class="profile-links-title">{link.title}</div>
        <ul class="profile-links-menu">
          {link.items.map(item => {
            return this.renderLi(item);
          })}
        </ul>
      </span>
    );
  }

  renderLi(item: { title: string, handler: () => void }): JSX.Element {
    return (
      <span>
      <li class="profile-links-menu-item" onClick={() => this.onClicked(item)}>
        {item.title} <i class="yo-right"></i>
      </li>
      <div class="border"></div>
      </span>
    );
  }
}
