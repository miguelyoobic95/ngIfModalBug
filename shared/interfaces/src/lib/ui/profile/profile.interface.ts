export interface IProfileConfig {
    user: any;
    links: Array<{
        title: string;
        items: Array<{
            title: string;
            handler: () => void;
        }>
    }>;
    logoutText?: string;
    hideLogout?: boolean;
}