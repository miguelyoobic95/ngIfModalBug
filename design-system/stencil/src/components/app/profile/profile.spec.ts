import { TestWindow } from '@stencil/core/testing';
import { YooProfileComponent } from './profile';

describe('YooProfileComponent', () => {
    const data = {
        user: { firstName: 'Cecilia', lastName: 'Adams', role: 'Store Manager', imageData: 'https://res.cloudinary.com/www-yoobic-com/image/upload/w_1020,h_480,c_fill,q_auto:low,f_auto,fl_lossy/a_exif/v1522152485/k7dedr96ainrdhmkzlgj.png' }, // might have to change the names
        links: [{
            title: 'More',
            items: [
                {
                    title: 'Documents',
                    handler: () => { alert('Documents'); }
                },
                {
                    title: 'Contacts',
                    handler: () => { alert('Contacts'); }
                },
                {
                    title: 'Notes',
                    handler: () => { alert('Notes'); }
                }
            ]
        },
        {
            title: 'Get started',
            items: [
                {
                    title: 'What\'s New?',
                    handler: () => { }
                },
                {
                    title: 'Walkthrough',
                    handler: () => { }
                }
            ]
        },
        {
            title: 'Support & Preferences',
            items: [
                {
                    title: 'Settings',
                    handler: () => { }
                },
                {
                    title: 'Help Center',
                    handler: () => { }
                },
                {
                    title: 'Terms & Conditions',
                    handler: () => { }
                }
            ]
        }
        ],
        logoutText: 'Log Out',
        hideLogout: false
    };
    it('should build', () => {
        expect(new YooProfileComponent()).toBeTruthy();
    });

    describe('Rendering', () => {
        let element, window;
        beforeEach(async () => {
            window = new TestWindow();
        });
        it('Should render', async () => {
            element = await window.load({
                components: [YooProfileComponent],
                html: '<yoo-profile></yoo-profile>'
            });
            expect(element).toMatchSnapshot();
        });
        it('Should render with config', async () => {
            element = await window.load({
                components: [YooProfileComponent],
                html: '<yoo-profile></yoo-profile>'
            });
            element.config = data;
            await window.flush();
            expect(element).toMatchSnapshot();
        });
    });
    describe('Events', () => {
        let element, emitEvent, window, res;
        beforeEach(async () => {
            window = new TestWindow();
            emitEvent = jest.fn(ev => { res = ev.detail; });
            element = await window.load({
                components: [YooProfileComponent],
                html: '<yoo-profile></yoo-profile>'
            });
            element.config = data;
            await window.flush();
        });
        xit ('should emit profileEdit on Listen', async () => {
            window.document.addEventListener('profileEdit', emitEvent);
            //todo
            expect(emitEvent).toHaveBeenCalled();
            expect(res).toEqual(true);
        });
        it('should emit clicked on click', async () => {
            window.document.addEventListener('clicked', emitEvent);
            let item = element.querySelector('.profile-links-menu-item');
            await item.click();
            expect(emitEvent).toHaveBeenCalled();
            expect(res.title).toEqual('Documents');
        });
        it('should emit logout on click', async () => {
            window.document.addEventListener('logout', emitEvent);
            let logout = element.querySelector('.profile-logout');
            await logout.click();
            expect(emitEvent).toHaveBeenCalled();
            expect(res).toEqual(true);
        });
    });
});