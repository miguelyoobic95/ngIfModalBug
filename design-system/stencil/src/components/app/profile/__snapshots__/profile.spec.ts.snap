// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooProfileComponent Rendering Should render 1`] = `
<yoo-profile
  class="hydrated"
  data-yoo-profile-host=""
>
  
  <div
    class="profile-content"
    data-yoo-profile=""
  >
    <div
      data-yoo-profile=""
    >
      loading
    </div>
  </div>
</yoo-profile>
`;

exports[`YooProfileComponent Rendering Should render with config 1`] = `
<yoo-profile
  class="hydrated"
  data-yoo-profile-host=""
>
  
  <div
    class="profile-content"
    data-yoo-profile=""
  >
    <span
      data-yoo-profile=""
    >
      <div
        class="profile-user"
        data-yoo-profile=""
      >
        <yoo-avatar
          bottom-right-icon="yo-pen"
          class="large gradient-success large-border"
          data-yoo-profile=""
        />
        <div
          class="profile-user-name"
          data-yoo-profile=""
        >
          Cecilia Adams
        </div>
        <div
          class="profile-user-role"
          data-yoo-profile=""
        >
          Store Manager
        </div>
      </div>
      <div
        class="profile-links"
        data-yoo-profile=""
      >
        <span
          data-yoo-profile=""
        >
          <div
            class="profile-links-title"
            data-yoo-profile=""
          >
            More
          </div>
          <ul
            class="profile-links-menu"
            data-yoo-profile=""
          >
            <span
              data-yoo-profile=""
            >
              <li
                class="profile-links-menu-item"
                data-yoo-profile=""
              >
                Documents 
                <i
                  class="yo-right"
                  data-yoo-profile=""
                />
              </li>
              <div
                class="border"
                data-yoo-profile=""
              />
            </span>
            <span
              data-yoo-profile=""
            >
              <li
                class="profile-links-menu-item"
                data-yoo-profile=""
              >
                Contacts 
                <i
                  class="yo-right"
                  data-yoo-profile=""
                />
              </li>
              <div
                class="border"
                data-yoo-profile=""
              />
            </span>
            <span
              data-yoo-profile=""
            >
              <li
                class="profile-links-menu-item"
                data-yoo-profile=""
              >
                Notes 
                <i
                  class="yo-right"
                  data-yoo-profile=""
                />
              </li>
              <div
                class="border"
                data-yoo-profile=""
              />
            </span>
          </ul>
        </span>
        <span
          data-yoo-profile=""
        >
          <div
            class="profile-links-title"
            data-yoo-profile=""
          >
            Get started
          </div>
          <ul
            class="profile-links-menu"
            data-yoo-profile=""
          >
            <span
              data-yoo-profile=""
            >
              <li
                class="profile-links-menu-item"
                data-yoo-profile=""
              >
                What's New? 
                <i
                  class="yo-right"
                  data-yoo-profile=""
                />
              </li>
              <div
                class="border"
                data-yoo-profile=""
              />
            </span>
            <span
              data-yoo-profile=""
            >
              <li
                class="profile-links-menu-item"
                data-yoo-profile=""
              >
                Walkthrough 
                <i
                  class="yo-right"
                  data-yoo-profile=""
                />
              </li>
              <div
                class="border"
                data-yoo-profile=""
              />
            </span>
          </ul>
        </span>
        <span
          data-yoo-profile=""
        >
          <div
            class="profile-links-title"
            data-yoo-profile=""
          >
            Support & Preferences
          </div>
          <ul
            class="profile-links-menu"
            data-yoo-profile=""
          >
            <span
              data-yoo-profile=""
            >
              <li
                class="profile-links-menu-item"
                data-yoo-profile=""
              >
                Settings 
                <i
                  class="yo-right"
                  data-yoo-profile=""
                />
              </li>
              <div
                class="border"
                data-yoo-profile=""
              />
            </span>
            <span
              data-yoo-profile=""
            >
              <li
                class="profile-links-menu-item"
                data-yoo-profile=""
              >
                Help Center 
                <i
                  class="yo-right"
                  data-yoo-profile=""
                />
              </li>
              <div
                class="border"
                data-yoo-profile=""
              />
            </span>
            <span
              data-yoo-profile=""
            >
              <li
                class="profile-links-menu-item"
                data-yoo-profile=""
              >
                Terms & Conditions 
                <i
                  class="yo-right"
                  data-yoo-profile=""
                />
              </li>
              <div
                class="border"
                data-yoo-profile=""
              />
            </span>
          </ul>
        </span>
      </div>
      <div
        class="profile-logout"
        data-yoo-profile=""
      >
        <div
          class="border"
          data-yoo-profile=""
        />
        <span
          class="profile-logout-text"
          data-yoo-profile=""
        >
          Log Out
        </span>
        <div
          class="border"
          data-yoo-profile=""
        />
      </div>
    </span>
  </div>
</yoo-profile>
`;
